source/docs/advanced/theme/server-side-rendering.md

```diff
@@ -0,0 +1,120 @@
+
+To generate this HTML page, Front-Commerce will render the React application as a string. During this process, your React components will trigger Apollo queries that will "fetch" data from the GraphQL layer. It may hit remote APIs and finally generate the HTML content of your page from these data.
+
+**While we try to make this process as smooth as possible for developers so you don't have to worry about the technical details, there are some things that you should be aware of while creating your theme.**
+
+This section details everything frontend developers must keep in mind when authoring components in a SSR context
+
+## There is no `window` on the server!
+
+Even though it may seem obvious, **you are very likely to come across this problem at some point!** Browser APIs are not available during SSR. Keep it in mind when using them, and provide a graceful fallback.
+
+Sometimes, it could be as simple as a default value or returning earlier from your function without any side effect.
+
+Examples:
+
+```js
+const doSomethingWithIntersectionObserver = () => {
+  if (typeof window === "undefined" || !window.IntersectionObserver) {
+    return;
+  }
+  // …
+}
+```
+
+```js
+const initializePosition = setPosition => {
+  if (typeof navigator === "undefined" || !("geolocation" in navigator)) {
+    setPosition(DEFAULT_LATITUDE, DEFAULT_LONGITUDE);
+  } else {
+    navigator.geolocation.getCurrentPosition(position => {
+      setPosition(position.coords.latitude, position.coords.longitude);
+    });
+  }
+}
+```
+
+## `branchServerClient` at the rescue!
+
+There may be situations were displaying a totally different component during the server rendering might be relevant.
+
+For instance, instead of displaying a map centered on the user geolocation it might be better to render a placeholder (page skeleton) on the server and load the map with additional data during <abbr title="Client Side Rendering">CSR</abbr>. Another use case would be to avoid an unnecessary overhead by not rendering some components during SSR: a social media feed may not make sense on the server.
+
+Front-Commerce provides a `branchServerClient` <abbr title="Higher Order Component">HOC</abbr> that allow you to conditionally render a component or another one on server and client side.
+
+Example:
+
+```js
+import branchServerClient from "web/core/branchServerClient";
+
+const MyClientSideComponent => props => <div>Hi!</div>;
+
+export default branchServerClient(
+  () => () => null, // do not render anything during SSR
+  BaseComponent => BaseComponent // render the base component on the client
+)(MyClientSideComponent);
+```
```

Comment:
Not sure if it's worth mentioning this tip here but I know it helped some of our integrators:

To memorize which element renders on the server and which one renders on the client, refer to the method name: `branchServerClient`, thus the first parameter is for the server and the second one is for the client.